packages/core-ethereum/src/commitment.ts

```diff
@@ -57,14 +70,45 @@ async function createCommitmentChain(db: HoprDB, channelId: Hash, setChainCommit
   log('commitment chain initialized')
 }
 
+/**
+ * Simple class encapsulating channel information
+ * used to generate the initial channel commitment.
+ */
+export class ChannelCommitmentInfo {
+  constructor(
+    public readonly channelEntry: ChannelEntry,
+    public readonly chainId: number,
+    public readonly contractAddress: string
+  ) {}
+
+  /**
+   * Generate the initial commitment seed using this channel information and the given
+   * private node key.
+   * @param nodePrivateKey Local node private key.
+   */
+  public createInitialCommitmentSeed(nodePrivateKey: Uint8Array): Uint8Array {
+    const channelSeedInfo = u8aConcat(
+      this.channelEntry.channelEpoch.serialize(),
+      toU8a(this.chainId, 4),
+      this.channelEntry.getId().serialize(),
+      toUtf8Bytes(this.contractAddress)
+    )
+
+    return deriveCommitmentSeed(nodePrivateKey, channelSeedInfo)
+  }
+}
+
 export async function initializeCommitment(
   db: HoprDB,
-  channelId: Hash,
+  nodePrivateKey: Uint8Array,
```

Comment:
I'd rather use the `PeerId` instance to hold private key since we use it everywhere else

Reply:
Seems like a good idea!